shtdlib.sh

```diff
@@ -157,6 +157,24 @@ function debug {
     fi
 }
 
+# Umask decorator, changes the umask for a function
+# To use this add a line like the following (without #) as the first line of a function
+# umask_decorator "${FUNCNAME[0]}" "${@}" && return
+function umask_decorator {
+    if [ "${FUNCNAME[0]}" != "${FUNCNAME[2]}" ] ; then
+        mask=0007
```

Comment:
Not sure if you can pass in parameters with the decorators but might be useful to have the mask value configurable.